docs/secret_management.md

```diff
@@ -43,113 +101,84 @@ spec:
-
-```console
-$ helm [install|upgrade] dd-operator --set "secretBackend.command=/my-secret-backend.sh" --set "image.repository=datadog-operator-with-secret-backend" ./chart/datadog-operator
-success
-```
+**Note**: Requires Datadog Operator v0.5.0+.
 
-#### Using the secret helper
+Kubernetes supports exposing Secrets as files inside a pod. Datadog provides a helper script in the Datadog Operator image to read the Secrets from files.
 
-Kubernetes supports exposing Secrets as files inside a pod, and we provide a helper script in the Datadog Operator image to read the Secrets from files.
+1. Mount the Secret in the Operator container. For example, at `/etc/secret-volume`. 
 
-First, mount the Secret in the Operator container, for instance at `/etc/secret-volume`. Then install or update the Datadog Operator deployment with the `.Values.secretBackend.command` parameter set to `/readsecret.sh` and the `.Values.secretBackend.arguments` parameter set to `/etc/secret-volume`.
+2. Install or update the Datadog Operator deployment with the `.Values.secretBackend.command` parameter set to `/readsecret.sh` and the `.Values.secretBackend.arguments` parameter set to `/etc/secret-volume`.
 
-**Note:** This secret helper requires Datadog Operator v0.5.0+
+### Deploying Agent components using the secret backend feature with the DatadogAgent 
 
-### How to deploy Agent components using the secret backend feature with the DatadogAgent (Operator 1.11+)
+**Note**: Requires Datadog Operator v1.11+.
 
-If using a custom script, create a Datadog Agent (or Cluster Agent) image following the example for the Datadog Operator above, and specify credentials using `ENC[<placeholder>]`.
+1. If you are using a custom script, create a Datadog Agent (or Cluster Agent) image and specify credentials using `ENC[<placeholder>]`:
 
-```yaml
-apiVersion: datadoghq.com/v2alpha1
-kind: DatadogAgent
-metadata:
-  name: datadog
-spec:
-  global:
-    credentials:
-      apiKey: ENC[<api-key-secret-id>]
-      appKey: ENC[<app-key-secret-id>]
-  # ...
-```
+   ```yaml
+   apiVersion: datadoghq.com/v2alpha1
+   kind: DatadogAgent
+   metadata:
+     name: datadog
+   spec:
+     global:
+       credentials:
+         apiKey: ENC[<api-key-secret-id>]
+         appKey: ENC[<app-key-secret-id>]
+     # ...
+   ```
 
-The secret backend command can be specified in the `spec.global.secretBackend.command`:
+2. Specify the secret backend command in `spec.global.secretBackend.command`:
 
-```yaml
-apiVersion: datadoghq.com/v2alpha1
-kind: DatadogAgent
-metadata:
-  name: datadog
-spec:
-  global:
-    secretBackend:
-      command: "/my-secret-backend.sh"
-  # ...
-```
+   ```yaml
+   apiVersion: datadoghq.com/v2alpha1
+   kind: DatadogAgent
+   metadata:
+     name: datadog
+   spec:
+     global:
+       credentials:
+         apiKey: ENC[<api-key-secret-id>]
+         appKey: ENC[<app-key-secret-id>]
+       secretBackend:
+         command: "/my-secret-backend.sh"
```

Comment:
I think it might be clearer if we actually consolidate 1. and 2. here in a single example

Then we can distinguish using a custom script (here) with the convenience script provided (below)

What do you think?